diff --git a/c/planarityCommandLine.c b/c/planarityCommandLine.c
index e714993..bfaaa01 100644
--- a/c/planarityCommandLine.c
+++ b/c/planarityCommandLine.c
@@ -6,8 +6,6 @@ See the LICENSE.TXT file for licensing information.
 
 #include "planarity.h"
 
-#include <unistd.h>
-
 int runQuickRegressionTests(int argc, char *argv[]);
 int callRandomGraphs(int argc, char *argv[]);
 int callSpecificGraph(int argc, char *argv[]);
