diff --git a/c/planarity.h b/c/planarity.h
index b0cfa62..dd6fc68 100644
--- a/c/planarity.h
+++ b/c/planarity.h
@@ -38,7 +38,17 @@ int commandLine(int argc, char *argv[]);
 int legacyCommandLine(int argc, char *argv[]);
 int menu();
 
-char Mode,
+#if defined(_MSC_VER)
+#if defined(PLANARITY_BUILD_DLL)
+#define PLANARITY_DLL __declspec(dllexport)
+#else
+#define PLANARITY_DLL __declspec(dllimport)
+#endif
+#else
+#define PLANARITY_DLL
+#endif
+
+PLANARITY_DLL extern char Mode,
      OrigOut,
      EmbeddableOut,
      ObstructedOut,
@@ -49,7 +59,7 @@ void Reconfigure();
 
 /* Low-level Utilities */
 #define MAXLINE 1024
-char Line[MAXLINE];
+PLANARITY_DLL extern char Line[MAXLINE];
 
 void Message(char *message);
 void ErrorMessage(char *message);
